packages/schema/src/language-server/zmodel-linker.ts

```diff
@@ -278,9 +278,11 @@ export class ZModelLinker extends DefaultLinker {
                 const model = getContainingModel(node);
 
                 if (model) {
-                    const userModel = getAllDeclarationsFromImports(this.langiumDocuments(), model).find(
-                        (d) => isDataModel(d) && d.name === 'User'
-                    );
+                    const dataModel = this.getContainingDataModel(node);
+                    const customAuthModel = dataModel && getModelAttributeValue('@@auth', dataModel);
```

Comment:
If we declare the attribute without a parameter, we can simply find it by checking attribute name. There's a `hasAttribute` helper from `@zenstackhq/sdk` for doing this.